tests/views/api_tests.py

```diff
@@ -500,3 +506,90 @@ def test_get_resource_jsonld_invalid_no_ontology(self):
+        self.assertEqual(len(response.json()), 2)
+        self.assertEqual(response.json()[0]["name"], "State 1")
+        self.assertEqual(response.json()[1]["name"], "State 2")
+
+
+class ResourceInstanceLifecycleStateTest(ArchesTestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(username="testuser", password="testpass")
+        self.lifecycle_state_id = uuid.uuid4()
+        self.new_lifecycle_state_id = uuid.uuid4()
+        self.resource_instance_id = uuid.uuid4()
+        self.resource_id = uuid.uuid4()
+
+    @patch("arches.app.models.models.ResourceInstance.objects.get")
+    def test_get_lifecycle_state(self, mock_get):
+        lifecycle_state = models.ResourceInstanceLifecycleState(
+            id=self.lifecycle_state_id, name="State 1"
+        )
+        mock_resource_instance = MagicMock(
+            resource_instance_lifecycle_state=lifecycle_state
+        )
+        mock_get.return_value = mock_resource_instance
+
+        response = self.client.get(
+            reverse(
+                "api_resource_instance_lifecycle_state",
+                args=[self.resource_instance_id],
+            )
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["name"], "State 1")
+
+    @patch("arches.app.models.resource.Resource.objects.get")
+    @patch("arches.app.models.models.ResourceInstanceLifecycleState.objects.get")
+    @patch(
+        "arches.app.models.resource.Resource.update_resource_instance_lifecycle_state"
+    )
+    def test_post_lifecycle_state(
+        self, mock_update, mock_get_lifecycle_state, mock_get_resource
+    ):
+        self.client.login(username="testuser", password="testpass")
+
+        original_lifecycle_state = models.ResourceInstanceLifecycleState(
+            id=self.lifecycle_state_id, name="State 1"
+        )
+        new_lifecycle_state = models.ResourceInstanceLifecycleState(
+            id=self.new_lifecycle_state_id, name="State 2"
+        )
+
+        # Configure the mock to return actual instances
+        mock_get_resource.return_value = Resource(
+            pk=self.resource_id,
+            resource_instance_lifecycle_state=original_lifecycle_state,
+        )
+        mock_get_lifecycle_state.return_value = new_lifecycle_state
+        mock_update.return_value = new_lifecycle_state
+
+        response = self.client.post(
+            reverse("api_resource_instance_lifecycle_state", args=[self.resource_id]),
+            data=json.dumps(str(self.new_lifecycle_state_id)),
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.json()["original_resource_instance_lifecycle_state"]["name"],
+            "State 1",
+        )
+        self.assertEqual(
+            response.json()["current_resource_instance_lifecycle_state"]["name"],
+            "State 2",
+        )
```

Comment:
Suggest rewriting this to just assert the shape of the response, not the values, since the values are just mocked. Thanks for writing a separate unit test to cover the correct updating behavior.